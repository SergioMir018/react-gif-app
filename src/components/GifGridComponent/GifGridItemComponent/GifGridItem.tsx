import React from 'react'
import PropTypes from 'prop-types'

import { GifImage} from '../GifGrid'

import '../../../styles/GifGridItem.css'

export const GifGridItem = ( { gif }: { gif: GifImage } ) => {

  const { title, url } = gif;

  return (
    <div className='card'>
      <img src={ url } alt={ title } />
      <p> { title } </p> 
    </div>
  )
}

GifGridItem.propTypes = {
  gif: PropTypes.shape({
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
}).isRequired
};