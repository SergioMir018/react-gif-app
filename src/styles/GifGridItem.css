.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 3px dashed rgba(109, 125, 150, 0.644);
  border-radius: 5px;
  overflow: hidden;
  margin-bottom: 10px;
  margin-right: 10px;
}

.card p {
  padding: 5px;
  text-align: center;
}

.card img {
  max-height: 190px;
  margin-bottom: 10px;
}