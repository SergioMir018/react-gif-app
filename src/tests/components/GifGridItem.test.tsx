import React from "react";
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridComponent/GifGridItemComponent/GifGridItem';

describe('Tests in <GifGridItem />', () => {
  const gif = { 
    title: 'Test GIF', 
    url: 'https://example.com/image.gif', 
    id: 'some-id' 
  };
  
  const wrapper = shallow(<GifGridItem gif={gif} />);

  test('should show the component correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('should have a paragraph in the title', () => {
    const paragraph = wrapper.find('p');

    expect(paragraph.text().trim()).toBe( gif.title );  
  });
  
  test('should have an image equal to the url and alt of the props', () => {
    const img = wrapper.find('img');
    
    const { src, alt } = img.props();

    expect( src ).toBe( gif.url );
    expect( alt ).toBe( gif.title );
  });
});

