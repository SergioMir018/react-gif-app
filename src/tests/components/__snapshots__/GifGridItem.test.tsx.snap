// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> should show the component correctly 1`] = `
<div
  className="card"
>
  <img
    alt="Test GIF"
    src="https://example.com/image.gif"
  />
  <p>
     
    Test GIF
     
  </p>
</div>
`;
